### ISSUE TITLE
Qt6 Webengine doesn't save cookies and caches

### ISSUE THREAD

**Issue by MeisterP** [GoldenCheetah/GoldenCheetah#4484]
* While working on a python chart accessing map tiles from a tile server, I've noticed that caching isn't working
* Logins in webpage charts aren't remembered either.

Turns out, Webengine in Qt6 no longer creates a Default Profile and therefore neither caches nor cookies are stored.

https://forum.qt.io/topic/131277/qtwebengine-cannot-save-cache
https://doc-snapshots.qt.io/qt6-dev/qtwebengine-changes-qt6.html#default-profile

**Comment by liversedge** [comment]
This is definitely a preference for us- would rather not introduce unneccesary vulnerabilities and attack surfaces.

**Comment by amtriathlon** [comment]
I agree this is a better behavior for OAuth, for the Web Page chart I think we could make it opt in, requiring explicit enable in chart config since cookies are useful for automatic login, for example to download original files from Garmin Connect. 
No sure about the 3 maps and Python web chart.

**Comment by liversedge** [comment]
I think you are right- we should look to enable caching for the map.

**Comment by amtriathlon** [comment]
Agree, we could create a different QWebEngineProfile to enable caching without persistent cookies for the maps using  setPersistentCookiesPolicy(QWebEngineProfile::NoPersistentCookies)
